### PR TITLE
docs(intercept): document matchUrlAgainstPath

### PR DESCRIPTION
**Pull request by flotwig**
For https://github.com/cypress-io/cypress/pull/14241

### CODE REVIEW

File: source/api/commands/intercept.md [hunk 1]
```diff
@@ -82,6 +82,11 @@ All properties are optional. All properties that are set must match for the rout
    * If 'false', only HTTP requests will be matched.
    */
   https?: boolean
+  /**
+   * If `true`, will match the supplied `url` against incoming `path`s.
+   * Cannot be set without `url` or with `path`.
```
Comment (jennifer-shehane) [inline source/api/commands/intercept.md:87]:
```suggestion
   * Requires a `url` argument. Cannot be used with a `path` argument.
```

File: source/api/commands/intercept.md [hunk 2]
```diff
@@ -159,6 +164,19 @@ The `routeHandler` defines what will happen with a request if the {% urlHash "`r
 
 ## Matching URL
 
+{% note info %}
+**Note:** passing a URL as a string or RegExp to `cy.intercept()` will automatically set `matchUrlAgainstPath` to `true`. This means that the supplied string or RegExp will be matched against the **path** if matching against the **URL** fails.
+{% endnote %}
```
Comment (jennifer-shehane) [inline source/api/commands/intercept.md:169]: I moved the note to the top of the section and combined them.

File: source/api/commands/intercept.md [hunk 3]
```diff
@@ -159,6 +164,19 @@ The `routeHandler` defines what will happen with a request if the {% urlHash "`r
...
+
+You can provide the entire URL to match
+
+```js
+// will match any request that exactly matches the URL
+//   matches GET https://prod.cypress.io/users
+//   won't match GET https://staging.cypress.io/users
+cy.intercept('https://prod.cypress.io/users')
+```
```
Comment (jennifer-shehane) [inline source/api/commands/intercept.md:178]: Added this example since there wasn't a full URL example.

**Review by jennifer-shehane — COMMENTED** [review]
Might want to add an example for `matchUrlAgainstPath: false`. I'm not sure how useful you see this being though, so that's up to you. 

Will need matchUrlAgainstPath in the History table also. I assume it'll get into 6.2.0.

**Comment by flotwig** [comment]
> Might want to add an example for matchUrlAgainstPath: false. I'm not sure how useful you see this being though, so that's up to you.

I see `matchUrlAgainstPath` as something that users will be using implicitly 99% of the time, not explicitly, so I don't feel that an extra example is needed.

**Review by jennifer-shehane — APPROVED** [review]

**Pull request merged**